--- a/configure.ac	2017-07-06 06:57:02.138151321 +0200
+++ b/configure.ac	2017-07-06 07:01:33.440159466 +0200
@@ -177,13 +177,14 @@
     AC_DEFINE([WITH_GLEW], [1], [Define to 1 to use GLEW])
 ])
 
+have_libmpeg2=no
 AC_ARG_WITH(libmpeg2, AS_HELP_STRING(
     [--without-libmpeg2], [or --with-libmpeg=builtin to use included version]))
 AM_CONDITIONAL([BUILTIN_LIBMPEG2], [test x$with_libmpeg2 = xbuiltin])
 AS_IF([test "x$with_libmpeg2" != xno], [
     AS_IF([test "x$with_libmpeg2" = xbuiltin], [
     ], [
-        PKG_CHECK_MODULES([LIBMPEG2], [libmpeg2 libmpeg2convert])
+        PKG_CHECK_MODULES([LIBMPEG2], [libmpeg2 libmpeg2convert], [have_libmpeg2=yes])
     ])
     AC_DEFINE([WITH_LIBMPEG2], [1], [Define to 1 to use libmpeg2])
 ])
@@ -750,7 +751,11 @@
 
 AC_DEFINE([FPU_UAE], [1], [Define to 1])
 
-AC_CONFIG_SUBDIRS([libmpeg2])
+AS_IF([test "x$have_libmpeg2" != xyes], [
+    AS_IF([test "x$with_libmpeg2" = xbuiltin], [
+        AC_CONFIG_SUBDIRS([libmpeg2])
+    ])
+])
 
 # Write Makefile
 
