diff -ru work.orig/src/osd/sdl/sdl.mak work/src/osd/sdl/sdl.mak
--- work.orig/src/osd/sdl/sdl.mak	2015-03-11 18:42:22.102068698 -0700
+++ work/src/osd/sdl/sdl.mak	2015-03-11 18:43:24.594579570 -0700
@@ -855,6 +855,7 @@
 BASELIBS += -lX11
 endif
 
+ifneq ($(NO_DEBUGGER),1)
 # The newer debugger uses QT
 ifndef NO_USE_QTDEBUG
 QTCFLAGS := $(shell pkg-config --cflags QtGui)
@@ -863,6 +864,7 @@
 INCPATH += $(QTCFLAGS)
 LIBS += $(QTLIBS)
 endif
+endif # NO_DEBUGGER
 
 # some systems still put important things in a different prefix
 LIBS += -L/usr/X11/lib -L/usr/X11R6/lib -L/usr/openwin/lib

