--- subversion-1.14.0-r1.ebuild	2021-01-04 23:08:18.405167049 +0800
+++ subversion-1.14.0-r2.ebuild	2021-01-04 23:05:22.024445768 +0800
@@ -1,7 +1,7 @@
 # Copyright 1999-2020 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=6
+EAPI=7
 
 PYTHON_COMPAT=( python3_{7,8} )
 USE_RUBY="ruby26 ruby25 ruby24"
@@ -9,7 +9,8 @@
 WANT_AUTOMAKE="none"
 GENTOO_DEPEND_ON_PERL="no"
 
-inherit autotools bash-completion-r1 db-use depend.apache flag-o-matic java-pkg-opt-2 libtool perl-module python-any-r1 ruby-single toolchain-funcs xdg-utils
+inherit autotools bash-completion-r1 db-use depend.apache flag-o-matic java-pkg-opt-2
+inherit libtool perl-module prefix python-any-r1 ruby-single toolchain-funcs xdg-utils
 
 MY_P="${P/_/-}"
 DESCRIPTION="Advanced version control system"
@@ -21,7 +22,7 @@
 LICENSE="Apache-2.0 BSD MIT BSD-2 FSFAP unicode"
 SLOT="0"
 [[ "${PV}" = *_rc* ]] || \
-KEYWORDS="~alpha amd64 arm arm64 ~hppa ~ia64 ppc ppc64 sparc x86 ~amd64-linux ~x86-linux"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~ppc ~ppc64 ~sparc ~x86 ~amd64-linux ~x86-linux"
 IUSE="apache2 berkdb debug doc extras gnome-keyring java kwallet nls perl ruby sasl test"
 RESTRICT="!test? ( test )"
 
@@ -162,6 +163,8 @@
 	sed -i -e '1c\#!/usr/bin/env sh' build/transform_libtool_scripts.sh || \
 		die "/bin/sh is not POSIX shell!"
 
+	hprefixify build/ac-macros/svn-macros.m4
+
 	eautoconf
 	elibtoolize
 
@@ -206,27 +209,12 @@
 	fi
 
 	case ${CHOST} in
-		*-aix*)
-			# avoid recording immediate path to sharedlibs into executables
-			append-ldflags -Wl,-bnoipath
-		;;
-		*-cygwin*)
-			# no LD_PRELOAD support, no undefined symbols
-			myconf+=( --disable-local-library-preloading LT_LDFLAGS=-no-undefined )
-			;;
-		*-interix*)
-			# loader crashes on the LD_PRELOADs...
-			myconf+=( --disable-local-library-preloading )
-		;;
 		*-solaris*)
 			# need -lintl to link
 			use nls && append-libs intl
 			# this breaks installation, on x64 echo replacement is 32-bits
 			myconf+=( --disable-local-library-preloading )
 		;;
-		*-mint*)
-			myconf+=( --enable-all-static --disable-local-library-preloading )
-		;;
 		*)
 			# inject LD_PRELOAD entries for easy in-tree development
 			myconf+=( --enable-local-library-preloading )
