--- a/Makefile	2017-07-06 14:58:57.292835593 +0200
+++ b/Makefile	2017-07-06 15:11:42.309858561 +0200
@@ -3,8 +3,8 @@
 dist_name = fs-uae-launcher-$(version)
 dist_dir := $(build_dir)/$(dist_name)
 
-DESTDIR :=
-prefix := /usr/local
+DESTDIR ?=
+prefix := $(PREFIX)
 
 ifeq ($(wildcard OpenGL),)
 	OpenGL_dir := "."
@@ -118,9 +118,6 @@
 	mkdir -p $(DESTDIR)$(prefix)/share
 	cp -a share/* $(DESTDIR)$(prefix)/share
 
-	mkdir -p $(DESTDIR)$(prefix)/share/doc/fs-uae-launcher
-	cp -a README COPYING $(DESTDIR)$(prefix)/share/doc/fs-uae-launcher
-
 install: install-data install-program
 
 dist_dir := fs-uae-launcher-$(version)
