--- a/src/xoj-preview-extractor/CMakeLists.txt.orig	2022-02-19 21:27:20.450726906 +0100
+++ b/src/xoj-preview-extractor/CMakeLists.txt	2022-02-19 21:29:23.372199989 +0100
@@ -41,10 +41,6 @@
 
 set (THUMBNAILER_BIN "xournalpp-thumbnailer")
 
-add_custom_command (TARGET xournalpp-thumbnailer POST_BUILD
-  COMMAND ${CMAKE_STRIP} ${THUMBNAILER_BIN}
-)
-
 ## Install ##
 
 install (TARGETS xournalpp-thumbnailer
