diff -up firefox-87.0/toolkit/crashreporter/breakpad-client/linux/handler/exception_handler.cc.old firefox-87.0/toolkit/crashreporter/breakpad-client/linux/handler/exception_handler.cc
--- firefox-87.0/toolkit/crashreporter/breakpad-client/linux/handler/exception_handler.cc.old	2021-03-25 19:07:12.920421009 +0100
+++ firefox-87.0/toolkit/crashreporter/breakpad-client/linux/handler/exception_handler.cc	2021-03-25 19:06:45.496596019 +0100
@@ -149,7 +149,7 @@ void InstallAlternateStackLocked() {
   // SIGSTKSZ may be too small to prevent the signal handlers from overrunning
   // the alternative stack. Ensure that the size of the alternative stack is
   // large enough.
-  static const unsigned kSigStackSize = std::max(16384, SIGSTKSZ);
+  static const unsigned kSigStackSize = (16384 > SIGSTKSZ) ? 16384 : SIGSTKSZ;
 
   // Only set an alternative stack if there isn't already one, or if the current
   // one is too small.
